pints/_mcmc/_adaptive.py

```diff
@@ -26,27 +28,190 @@ class AdaptiveCovarianceMCMC(pints.MCMC):
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_burn_in_rate(self, rate):
+        """
+        Set the first fraction of the total number of iterations to discard as 
+        burn-in.
+        """
+        self._burn_in_rate = rate
+        self._burn_in = int(self._iterations * self._burn_in_rate)
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_burn_in(self, burn_in):
+        """
+        Set the number of iterations to discard as burn-in.
+        """
+        self._burn_in = burn_in
+        self._burn_in_rate = self._burn_in/self._iterations
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_thinning(self, thinning):
+        """
+        Set thinning - store only one sample per X
+        """
+        self._thinning = thinning
+        if self._thinning > self._adaptation:
+            print('WARNING: Thinning larger than adaptation')
+        if self._thinning > self._burn_in:
+            print('WARNING: Thinning larger than burn-in')
+        if self._thinning > self._iterations:
+            raise Exception('Thinning larger than total number of iterations')
+
+    def print_setup(self):
+        """
+        Print to console the current setup info of the adaptive convariance
+        MCMC routine.
+        """
+        print('\n## Adaptive convariance MCMC routine setup info')
+        print('Target acceptance rate: ' + str(self._acceptance_target))
+        print('Total number of iterations: ' + str(self._iterations))
+        print('Number of iterations before adapation: ' + str(self._adaptation))
+        print('Number of iterations to discard as burn-in: ' + str(self._burn_in))
+        print('Thinning: Store only one sample per ' + str(self._thinning) + '\n')
+
     def run(self):
         """See: :meth:`pints.MCMC.run()`."""
 
+        # Problem dimension
+        d = self._dimension
+
         # Target acceptance rate
-        acceptance_target = 0.25
+        acceptance_target = self._acceptance_target
 
         # Total number of iterations
-        #TODO Allow changing before run() with method call
-        iterations = 2000 * self._dimension
+        iterations = self._iterations
 
         # Number of iterations before adapation
-        #TODO Allow changing before run() with method call
-        adaptation = int(iterations * 0.25)
+        adaptation = self._adaptation
 
         # Number of iterations to discard as burn-in
-        #TODO Allow changing before run() with method call
-        burn_in = int(iterations * 0.5)
+        burn_in = self._burn_in
 
         # Thinning: Store only one sample per X
-        #TODO Allow changing before run() with method call
-        thinning = 4
+        thinning = self._thinning
+
+        # Print out what we are using once
+        self.print_setup()
```

Comment:
This is ok, but maybe just copy-paste the code you have for print_setup into this method.

Reply:
This is also the place where you should check the argument. Raise a ValueError if they're wrong.